File: Lab1/Person.cpp
#include "person.h"
#include "company.h"

Person::Person(const std::string& Surname, const std::string& Name, const std::string& Patronymic) {
    this->Name = Name;
    this->Surname = Surname;
    this->Patronymic = Patronymic;
}

const std::string& Person::GetNamePerson() const {
    return Name;
}

const std::string& Person::GetSurnamePerson() const {
    return Surname;
}

const std::string& Person::GetPatronymicPerson() const {
    return Patronymic;
}

uint16_t Person::GetCountCompany() {
    return CompanyCount;
}

bool Person::JoinACompany(Company& company) {
    if(!company.JoinAPerson(*this)) {
        CompanyCount++;

        return true;
    }

    else {
        return false;
    }
}


bool Person::LeaveCompany(Company& company) {
    if(!company.DismissPerson(*this)) {
        CompanyCount--;

        return true;
    }

    else {
        return false;
    }
}

Person::~Person() = default;

std::ostream& operator<<(std::ostream& stream, Person& person) {
    stream << "******************************************************" << std::endl;
    stream << "ФИО работника: " << person.GetSurnamePerson() << " " << person.GetNamePerson() << " " << person.GetPatronymicPerson() << std::endl;
    stream << "******************************************************" << std::endl;

    return stream;
}

File: Lab2/Person.java
package com.company;

public class Person {
    private String Name;
    private String Surname;
    private String Patronymic;

    //private  int CompanyCount = 0;

    public Person (String Surname, String Name, String Patronymic) {
        this.Surname = Surname;
        this.Name = Name;
        this.Patronymic = Patronymic;
    }

    public final String GetNamePerson () {
        return Name;
    }

    public final String GetSurnamePerson () {
        return Surname;
    }

    public final String GetPatronymicPerson () {
        return Patronymic;
    }

    /*public int GetCountCompany () {
        return CompanyCount;
    }*/

    @Override
    public String toString () {
        return "Фамилия работника = " + Surname
                + ", Имя работника = " + Name
                + ", Отчество работника = " + Patronymic;
    }

    public boolean compare(String surname, String name, String patronymic) {
        return this.Surname.equals(surname) && this.Name.equals(name) && this.Patronymic.equals(patronymic);
    }
}

